### DISCUSSION TITLE
Issue with Devtools and State Mutation Naming in TypeScript Project

### DISCUSSION THREAD

**Discussion by NativSibony** [pmndrs/zustand#2623] [Bug report] [answered]
### Bug Description

I'm utilizing devtools to differentiate between various instances of state updates/initializations. In the attached playground, you'll notice the problem I'm encountering: the set function doesn't anticipate the third argument meant to define the actual name of the mutation. While it appears to function correctly in the Redux Chrome extension, this discrepancy is hindering the progress of my project due to a TypeScript error.

### Reproduction Link
Here is the minimal reproduction with TypeScript Playground: [playground](https://stackblitz.com/edit/react-ts-qa3twj?file=slice.ts)

**Comment by dbritto-dev** [comment]
@NativSibony middlewares shouldn't be use on slices, middleware works at store level so that code is wrong

**Reply by NativSibony** [reply]
Fine, moved the middleware to correct position... 
The issue is still there and your comment did not fixed that :) @dbritto-dev

**Reply by dbritto-dev** [reply]
@NativSibony sounds right to me, due to missing types in your slice creator. I encourage you to read the TypeScript guide, here you go: https://docs.pmnd.rs/zustand/guides/typescript#common-recipes

**Comment by NativSibony** [comment, marked as answer]
My solution for those who encounter the same problem.
```tsx
type SliceCreator<T> = StateCreator<T, [['zustand/devtools', never]], []>;
```
